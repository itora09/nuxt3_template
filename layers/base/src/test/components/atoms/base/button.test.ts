import { describe, test, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import AtomsBaseButton from '../../../../components/atoms/base/button.vue'

describe('AtomsBaseButton', () => {
  test('snapshot', () => {
    const wrapper = mount(AtomsBaseButton)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('デフォルトでbuttonタグがレンダリングされる', () => {
    const wrapper = mount(AtomsBaseButton)
    expect(wrapper.find('button').exists()).toBe(true)
  })

  test('tagプロップスでaタグを指定できる', () => {
    const wrapper = mount(AtomsBaseButton, {
      props: {
        tag: 'a',
      },
    })
    expect(wrapper.find('a').exists()).toBe(true)
  })

  test('スロットの内容が正しく表示される', () => {
    const slotContent = 'ボタンテキスト'
    const wrapper = mount(AtomsBaseButton, {
      slots: {
        default: slotContent,
      },
    })
    expect(wrapper.text()).toBe(slotContent)
  })
})
