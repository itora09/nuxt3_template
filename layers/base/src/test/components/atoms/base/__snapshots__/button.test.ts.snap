// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AtomsBaseButton > snapshot 1`] = `
<button
  blank="false"
  class="atoms-base-button"
  data-v-6ee581dc=""
  forceanchorlink="false"
  nolocale="false"
  type="button"
>
  
  
</button>
`;
